src/services/sync/strategies/WebUntis/WebUntisSchoolyearSyncer.js

```diff
@@ -0,0 +1,503 @@
+const assert = require('assert');
+
+const WebUntisBaseSyncer = require('./WebUntisBaseSyncer');
+
+
+/**
+ * Implements syncing from WebUntis API based on the Syncer interface
+ * @class WebUntisSchoolyearSyncer
+ * @implements {Syncer}
+ */
+class WebUntisSchoolyearSyncer extends WebUntisBaseSyncer {
+	/**
+     * Constructor
+     *
+     * Disabled because of ESlint
+     *
+     * @param {*} app
+     * @param {*} stats
+     * @param {*} account
+	 * @param {boolean} dryrun
+     */
+	/* constructor(app, stats, account, dryrun) {
+		super(app, stats, account, dryrun);
+	} */
+
+	/**
+	 * @see {Syncer#respondsTo}
+	 */
+	static respondsTo(target) {
+		return target === 'webuntis-schoolyear';
+	}
+
+	static params(params /* , data */) {
+		return [params.account,
+			!("dryrun" in params.query) || params.query.dryrun !== 'false'
+		];
+	}
+
+	/**
+	 * @see {Syncer#steps}
+     *
+     * Steps:
+     * * Check for WebUntis system (may have configured none)
+     * * Login to WebUntis
+     * * Fetch Lesson/Teacher/Room/... changes
+     * * Migrate the changes
+     * * Generate events using text templates
+     * * Send out events for all affected student/teacher/...
+     *
+     * Assumptions:
+     * * Each school has one WebUntis system at most
+     * * This syncer has to be triggered for each school individually
+	 */
+	steps() {
+		this.logInfo(`Running WebUntis Sync in dry run: ${this.dryrun}.\n`);
+
+		/* Why not
+        return this.getWebUntisSystems().then(
+            systems => {
+                // ...
+            }
+        );
+        ? */
+		return this.getUser()
+			.then(user => this.getWebUntisSystems(user))
+			.then(([systems, school]) => {
+				/* TODO: Remove later */
+				// if (systems.length === 0) {
+				// Configure test environment until configuration is available via GUI
+				systems[0] = {
+					alias: 'Test WebUntis',
+					type: 'webuntis',
+					webuntisConfig: {
+						active: true,
+						url: '[URL]',
+						schoolname: '[schoolname]',
+						user: '[username]',
+						password: '[password]',
+					},
+				};
+				// }
+
+				if (systems.length === 0) {
+					return Promise.reject(
+						new Error('No WebUntis configuration for associated school.'),
+					);
+				}
+				this.logInfo(`Found ${systems.length} WebUntis configurations for school ${school.name}.\n`);
+				
+				return Promise.all(systems.map((system) => {
+					this.stats.systems[school.name] = {};
+					return this.syncFromSystem(system, this.stats.systems[school.name], school);
+				}));
+			});
+	}
+
+	async syncFromSystem(system, stats, school) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['school' is defined but never used. Allowed unused args must match /app|req|res|next|options|params/u.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=4061937)